src/daft-connect/src/spark_analyzer.rs

```
@@ -295,22 +294,86 @@ impl SparkAnalyzer<'_> {
+
+                if let Some(sep) = options.get("sep").and_then(|s| s.chars().next()) {
+                    builder = builder.delimiter(sep);
+                }
+
+                // spark sets this to false by default, so we'll do the same
+                let header = options
+                    .get("header")
+                    .and_then(|v| v.to_lowercase().parse().ok())
+                    .unwrap_or(false);
+
+                builder = builder.has_headers(header);
+
+                let infer_schema = options
+                    .get("inferSchema")
+                    .and_then(|v| v.to_lowercase().parse().ok())
+                    .unwrap_or(true);
+
+                builder = builder.infer_schema(infer_schema);
+
+                if let Some(quote) = options.get("quote").and_then(|s| s.chars().next()) {
+                    builder = builder.quote(quote);
+                }
+
+                if let Some(comment) = options.get("comment").and_then(|s| s.chars().next()) {
+                    builder = builder.comment(comment);
+                }
+
+                if let Some(escape_char) = options.get("escape").and_then(|s| s.chars().next()) {
+                    builder = builder.escape_char(escape_char);
+                }
+
+                if let Some(hive_partitioning) = options
+                    .get("hive_partitioning")
+                    .and_then(|v| v.parse().ok())
+                {
+                    builder = builder.hive_partitioning(hive_partitioning);
+                }
+                if let Some(chunk_size) = options.get("chunk_size").and_then(|v| v.parse().ok()) {
+                    builder = builder.chunk_size(chunk_size);
+                }
+
+                if let Some(buffer_size) = options.get("buffer_size").and_then(|v| v.parse().ok()) {
+                    builder = builder.buffer_size(buffer_size);
+                }
+
+                builder.finish().await?
```

Same goes for the other formats. Would we like to error when an option is not supported rather than silently ignore it? Perhaps this would result in feature requests rather than bug reports or possibly producing different results than the customer expects.